internal/goroutine/periodic.go

```diff
@@ -381,6 +382,21 @@ func (r *PeriodicGoroutine) withRecorder(ctx context.Context, f func(ctx context
 	return err
 }
 
+// runAndConvertPanicToError invokes f with the given ctx and recovers any panics
+// by turning them into an error instead.
+func runAndConvertPanicToError(ctx context.Context, f func(ctx context.Context) error) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			if e, ok := r.(error); ok {
+				err = errors.Wrap(e, "panic occurred")
+			} else {
+				err = fmt.Errorf("panic occurred: %v", r)
```

Comment:
I think our linters will yell at  you for using `fmt.Errorf`

```suggestion
				err = errors.Newf("panic occurred: %v", r)
```